pkg/api/graph/deployment.go

```diff
@@ -0,0 +1,415 @@
+package graph
+
+import (
+	"sort"
+
+	"github.com/gonum/graph"
+	"github.com/gonum/graph/search"
+
+	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+
+	//build "github.com/openshift/origin/pkg/build/api"
+	deploy "github.com/openshift/origin/pkg/deploy/api"
+	image "github.com/openshift/origin/pkg/image/api"
+)
+
+// DeploymentFlowMap describes a single deployment config and the objects
+// that contributed to that deployment.
+type DeploymentFlowMap map[*DeploymentConfigNode][]ImageDeploymentFlow
+
+type ImageDeploymentFlow struct {
```

Comment:
this would make more sense to me if it were called ImageBuildFlow or something similar, because it contains references to everything that leads to the creation of the image, and doesnt have any direct correlation to a deployment.